hopp/simulation/technologies/wind/floris.py

```diff
@@ -239,27 +265,90 @@ def export(self):
+        xcoords,ycoords = self.fi.get_turbine_layout()
+        return xcoords,ycoords
+    
+    def set_wind_farm_layout(self,xcoords,ycoords):
+        """Set wind farm layout coordinates and update system capacity and number
+            of turbines.
 
         Args:
-            xcoords (list[float]): A list of x-coordinates for turbine locations.
-            ycoords (list[float]): A list of y-coordinates for turbine locations.
+            xcoords (list[float]): x-coordinates of wind turbines in meters.
+            ycoords (list[float]): y-coordinates of wind turbines in meters.
 
         Raises:
-            ValueError: If x- and y-coordinates are not the same length, an error is raised.
+            ValueError: "WindPlant turbine coordinate arrays must have same length"
         """
         if len(xcoords) != len(ycoords):
             raise ValueError("WindPlant turbine coordinate arrays must have same length")
-        self.fi.set(layout_x=xcoords, layout_y=ycoords)
+        self.fi.set(
+            layout_x = xcoords,
+            layout_y = ycoords
+            )
         self.nTurbs = len(xcoords)
-        self.system_capacity = len(xcoords) * self.turb_rating
+        self.system_capacity = len(xcoords)*self.turb_rating
         self.value("wind_farm_xCoordinates", xcoords)
         self.value("wind_farm_yCoordinates", ycoords)     
+    
+    def initialize_wind_turbine(self, floris_config): 
+        """Update `floris_config` with turbine parameters. Checks the turbine library 
+            and floris internal library for a turbine with name matching either 
+            `config.turbine_name` or `floris_config["farm"]["turbine_type"][0]["turbine_type"]`.
+            If no matching turbine is found, find the turbine with the closest 
+            matching name and raise a warning.
+
+        Args:
+            floris_config (dict): floris (version 4) input dictionary.
+
+        Raises:
+            UserWarning: if turbine name does not exactly match a turbine in 
+                the turbine library or floris library.
+
+        Returns:
+            dict: floris config with turbine model parameters updated in `floris_config["farm"]["turbine_type"][0]`
+        """
+        if self.config.turbine_name is None:
+            
+            # turbine data is included in floris_config
+            if isinstance(floris_config["farm"]["turbine_type"][0],dict):
+                self.turbine_name = floris_config["farm"]["turbine_type"][0]["turbine_type"]
+
+            # load file from internal floris library
+            if isinstance(floris_config["farm"]["turbine_type"][0],str):
+                turbine_lib_res = floris_tools.check_libraries_for_turbine_name_floris(floris_config["farm"]["turbine_type"][0], self)
+                if isinstance(turbine_lib_res,str):
+                    raise UserWarning(turbine_lib_res)
+                floris_config["farm"]["turbine_type"][0] = turbine_lib_res
```

Comment:
`floris_config` could be returned after this line, and then the rest of the method could be dedented, and be without an `else` block.